### ISSUE TITLE
ghcWithPackages now uses ghc-8.2.2 regardless of what package set it's called from

### ISSUE THREAD

**Issue by pikajude** [NixOS/nixpkgs#33547]
```
$ nix-shell -p 'with import <nixpkgs> {}; pkgs.haskell.packages.ghc802.ghcWithPackages (p: [ p.wreq ])' --command 'ghc-pkg list | grep wreq && echo $?'
1
```

```
$ nix-build -E 'with import <nixpkgs> {}; pkgs.haskell.packages.ghc802.ghcWithPackages (p: [ p.wreq ])'
/nix/store/zcis4fhv0yaqacpj2zhlazd9zyxr7gaz-ghc-8.2.2-with-packages
```

**Comment by pikajude** [comment]
Looks to be caused by 5e31e828f88a8aa229d68bade00d9f2d70f7b9f8

**Comment by globin** [comment]
Closed by b2e9096
